File: cash.py
def get_change():
    while True:
        try:
            dollars = float(input("Change owed: "))
            if dollars >= 0:
                return round(dollars * 100)
        except ValueError:
            pass


def main():
    cents = get_change()
    coins = 0
    for value in (25, 10, 5, 1):
        coins += cents // value
        cents %= value
    print(coins)


if __name__ == "__main__":
    main()

File: cash.c
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        printf("Change owed: ");
        scanf("%f", &dollars);
    }
    while (dollars < 0);

    int cents = (int) round(dollars * 100);

    int coins = 0;
    int values[] = {25, 10, 5, 1};
    for (int i = 0; i < 4; i++)
    {
        coins += cents / values[i];
        cents %= values[i];
    }

    printf("%i\n", coins);
}
